android/src/main/java/com/robinpowered/RNMDMManager/RNMobileDeviceManagerModule.java

```diff
@@ -80,7 +125,51 @@ public void getConfiguration(final Promise promise) {
             }
             promise.resolve(data);
         } else {
-          promise.reject(new Error("Managed App Config is not supported"));
+            promise.reject(new Error("Managed App Config is not supported"));
+        }
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeSupported(final Promise promise) {
+        promise.resolve(isLockStatePermitted());
+    }
+
+    @ReactMethod
+    public void isSingleAppModeEnabled(final Promise promise) {
+        promise.resolve(isLockState());
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeEnabled(final Promise promise) {
+        boolean lockedStatus = false;
+        ActivityManager am = (ActivityManager) getReactApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                lockedStatus = am.getLockTaskModeState() == ActivityManager.LOCK_TASK_MODE_LOCKED;
+            } else {
+                lockedStatus = isLockStatePermitted() && am.isInLockTaskMode();
+            }
+        } catch (e) {
+            promise.reject(e);
+        }
+        promise.resolve(lockedStatus);
```

Comment:
This should happen in the `try` section, otherwise it'll attempt to fire if there's a caught exception